crates/core/src/factories/core_publish/message_handler.rs

```diff
@@ -0,0 +1,141 @@
+            PublishMessageType::Ops => {
+                let request =
+                    PublishOps::decode(publish.data).map_err(|err| {
+                        K2Error::other_src(
+                            format!("could not decode publish ops from {peer}"),
+                            err,
+                        )
+                    })?;
+                self.incoming_publish_ops_tx
+                    .try_send((request.into(), peer))
+                    .map_err(|err| {
+                        K2Error::other_src(
+                            "could not insert incoming publish ops request into queue",
+                            err,
+                        )
+                    })
+            }
+            unknown_message => Err(K2Error::other(format!(
+                "unknown publish message: {unknown_message:?}"
+            ))),
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::PublishMessageHandler;
+    use bytes::Bytes;
+    use kitsune2_api::*;
+    use kitsune2_test_utils::id::random_op_id;
+    use kitsune2_test_utils::space::TEST_SPACE_ID;
+    use prost::Message;
+    use std::time::Duration;
+
+    #[test]
+    fn decoding_error() {
+        let (incoming_publish_ops_tx, _) = tokio::sync::mpsc::channel(1);
+        let message_handler = PublishMessageHandler {
+            incoming_publish_ops_tx,
+        };
+        let peer = Url::from_str("wss://127.0.0.1:1").unwrap();
+        let wrong_message =
+            Bytes::from_static(b"this is not a publish message");
+        message_handler
+            .recv_module_msg(
+                peer,
+                TEST_SPACE_ID,
+                crate::factories::core_publish::PUBLISH_MOD_NAME.to_string(),
+                wrong_message,
+            )
+            .unwrap_err();
+    }
+
+    #[test]
+    fn invalid_message_type() {
+        let (incoming_publish_ops_tx, _) = tokio::sync::mpsc::channel(1);
+        let message_handler = PublishMessageHandler {
+            incoming_publish_ops_tx,
+        };
+        let peer = Url::from_str("wss://127.0.0.1:1").unwrap();
+        let request_message = K2FetchMessage {
+            fetch_message_type: 9,
+            data: Bytes::from_static(b"op"),
+        }
+        .encode_to_vec()
+        .into();
+
+        message_handler
+            .recv_module_msg(
+                peer,
+                TEST_SPACE_ID,
+                crate::factories::core_publish::PUBLISH_MOD_NAME.to_string(),
+                request_message,
+            )
+            .unwrap_err();
+    }
+
+    #[tokio::test]
+    async fn publish_ops() {
+        let (incoming_publish_ops_tx, mut incoming_publish_ops_rx) =
+            tokio::sync::mpsc::channel(1);
+        let message_handler = PublishMessageHandler {
+            incoming_publish_ops_tx,
+        };
+        let peer = Url::from_str("wss://127.0.0.1:1").unwrap();
+        let requested_op_ids = vec![random_op_id()];
```

Comment:
use `create_op_id_list`?

Reply:
funnily enough I copied that part of the code over from your fetch module :) - gonna change it in both places.

Reply:
addressed.